ganttproject/src/main/java/net/sourceforge/ganttproject/GanttProject.java

```diff
@@ -880,4 +895,21 @@ public void refresh() {
     super.repaint();
   }
 
+  private Unit sendProjectStateLogs() {
```

Comment:
It's not normal, is it? :)
Is it better to replace `this::sendProjectStateLogs` usage instead?

Reply:
returning Unit from Java is ok

Reply:
What is not so good is that this method will be triggered on literally every change in the tasks, which basically mean that every change will be sent to Colloboque as a separate log record. This may: 1) be inefficient and 2) break the [implied] transaction. Say, the user moved the project start to some new date. This will change the dates of all tasks, but from the user's perspective it is a single atomic transaction. 

This might be ok for now just for testing, but please add TODO to refactor it. We have so-called "undoable actions" which correspond to transactions from the user perspective (e.g. invoking a single action from the menu) and it is better to group xlof changes accumulated during such "transactions"

Reply:
Thanks for the remark! It should be definitely improved. Let's leave it as TODO for now, I suspect it requires quite a lot of investigation.